governance/xc_admin/packages/xc_admin_frontend/components/programs/PythCore.tsx

```diff
@@ -0,0 +1,505 @@
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const file = (e.target as HTMLInputElement).files![0]
+      const reader = new FileReader()
+      reader.onload = (e) => {
+        if (e.target) {
+          const fileData = e.target.result
+          try {
+            const uploadedConfig = JSON.parse(fileData as string)
+            const validation = validateUploadedConfig[ProgramType.PYTH_CORE](
+              data,
+              uploadedConfig,
+              cluster
+            )
+
+            if (!validation.isValid) {
+              toast.error(validation.error || 'Invalid configuration')
+              return
+            }
+
+            setDataChanges(validation.changes)
+            openModal()
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          } catch (error: any) {
+            toast.error(capitalizeFirstLetter(error.message))
+          }
+        }
+      }
+      reader.readAsText(file)
+    })
+    document.body.appendChild(uploadAnchor) // required for firefox
+    uploadAnchor.click()
+    uploadAnchor.remove()
+  }
+
+  const handleSendProposalButtonClick = () => {
+    const handleSendProposalButtonClickAsync = async () => {
+      setIsSendProposalButtonLoading(true)
+      if (pythProgramClient && dataChanges && !isMultisigLoading) {
+        try {
+          const multisigAuthority = readOnlySquads.getAuthorityPDA(
+            PRICE_FEED_MULTISIG[getMultisigCluster(cluster)],
+            1
+          )
+          const fundingAccount = isRemote
+            ? mapKey(multisigAuthority)
+            : multisigAuthority
+
+          // Generate instructions using the program registry functions
+          const instructions = await generateInstructions[
+            ProgramType.PYTH_CORE
+          ](dataChanges, cluster, {
+            fundingAccount,
+            pythProgramClient,
+            messageBufferClient,
+            connection,
+            rawConfig,
+          })
+
+          const response = await axios.post(
+            proposerServerUrl + '/api/propose',
+            {
+              instructions,
+              cluster,
+            }
+          )
+          const { proposalPubkey } = response.data
+          toast.success(`Proposal sent! 🚀 Proposal Pubkey: ${proposalPubkey}`)
+          setIsSendProposalButtonLoading(false)
+          closeModal()
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          if (error.response) {
+            toast.error(capitalizeFirstLetter(error.response.data))
+          } else {
+            toast.error(capitalizeFirstLetter(error.message))
+          }
+          setIsSendProposalButtonLoading(false)
+        }
+      }
+    }
+
+    handleSendProposalButtonClickAsync()
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const MetadataChangesRows = ({ changes }: { changes: any }) => {
```

Comment:
We should try very hard not to create components inside of closures, especially not in the render path of other components -- this will break React's ability to optimize the render tree as React will see this as a new function on each render pass and as such it has no way to know that the part of the subtree is not changing.

I recommend hoisting this component out to the top level of the module.

Same goes for the other components created inside this closure.

Also if we can avoid `any` here we should.